lib/post_action_creator.rb

```diff
@@ -342,32 +346,26 @@ def create_message_creator
       raw: body,
     }
 
-    if %i[notify_moderators spam illegal].include?(@post_action_name)
+    if @post_action_name == :notify_user
+      create_args[:subtype] = TopicSubtype.notify_user
+
+      create_args[:target_usernames] = @post.user.username
+
+      # Evaluate DiscoursePluginRegistry.post_action_notify_user_handlers.
+      # If any return false, return early from this method
+      handler_values =
+        DiscoursePluginRegistry.post_action_notify_user_handlers.map do |handler|
+          handler.call(@created_by, @post, @message)
+        end
+      return if handler_values.any? { |value| value == false }
+    else
       create_args[:subtype] = TopicSubtype.notify_moderators
       create_args[:target_group_names] = [Group[:moderators].name]
 
       if SiteSetting.enable_category_group_moderation? &&
            @post.topic&.category&.reviewable_by_group_id?
         create_args[:target_group_names] << @post.topic.category.reviewable_by_group.name
       end
-    else
-      create_args[:subtype] = TopicSubtype.notify_user
-
```

Comment:
So now in `else` block we always had `notify_user` which means that elsif with "x" was never happening.

That's why I could completely invert this condition and drop "x"